src/tui.rs

```diff
@@ -79,6 +82,9 @@ pub fn render_tui(key: String, proj_name: &str, submodules: &[Submodule]) {
                 }
                 KeyCode::Down => {
                     if let Some(selected) = template_list_state.selected() {
+                        let templates_length = parse_json::read_json()
+                            .expect("can fetch template list")
+                            .len();
                         let templates_length = submodules.len();
```

Comment:
Templates Length repeated again, just take care of repetition generally in the whole codebase